pandas/io/packers.py

```diff
@@ -115,8 +115,12 @@ def writer(fh):
 
     path_or_buf = _stringify_path(path_or_buf)
     if isinstance(path_or_buf, str):
-        with open(path_or_buf, mode) as fh:
-            writer(fh)
+        try:
+            with open(path_or_buf, mode) as fh:
+                writer(fh)
+        except FileNotFoundError:
+            raise ValueError('path_or_buf is invalid or was not found')
```

Comment:
Hmm not to overcomplicated things but shouldn't we still raise a FileNotFoundError? ValueError is inconsistent with other io methods

Reply:
yeah I think just re-raising FileNotFoundError with the new message might be helpful @U09Kane

Reply:
Sure thing